operator/v1/types.go

```diff
@@ -41,6 +45,65 @@ type ClusterManagerStatus struct {
 	// Degraded: components in hub do not match the desired configuration and only provide
 	// degraded service.
 	Conditions []StatusCondition `json:"conditions"`
+
+	// Generations are used to determine when an item needs to be reconciled or has changed in a way that needs a reaction.
+	// +optional
+	Generations []GenerationStatus `json:"generations,omitempty"`
+
+	// RelatedResources are used to track the resources that are related to this ClusterManager
+	// +optional
+	RelatedResources []RelatedResourceMeta `json:"relatedResources,omitempty"`
+}
+
+// RelatedResourceMeta represents the resource that is managed by an operator
+type RelatedResourceMeta struct {
+	// group is the group of the thing you're tracking
+	// +required
+	Group string `json:"group"`
+
+	// version is the version of the thing you're tracking
+	// +required
+	Version string `json:"version"`
+
+	// resource is the resource type of the thing you're tracking
+	// +required
+	Resource string `json:"resource"`
+
+	// namespace is where the thing you're tracking is
+	// +optional
+	Namespace string `json:"namespace"`
+
+	// name is the name of the thing you're tracking
+	// +required
+	Name string `json:"name"`
+}
+
+// GenerationStatus keeps track of the generation for a given resource so that decisions about forced updates can be made.
+// the definition matches the GenerationStatus defined in github.com/openshift/api/v1
+type GenerationStatus struct {
+	// group is the group of the thing you're tracking
+	// +required
+	Group string `json:"group"`
+
+	// version is the version of the thing you're tracking
+	// +required
+	Version string `json:"version"`
+
+	// resource is the resource type of the thing you're tracking
+	// +required
+	Resource string `json:"resource"`
+
+	// namespace is where the thing you're tracking is
+	// +optional
+	Namespace string `json:"namespace"`
+
+	// name is the name of the thing you're tracking
+	// +required
+	Name string `json:"name"`
+
+	// lastGeneration is the last generation of the controller involved
```

Comment:
I'm not sure what this is supposed to mean, exactly. Is it the last generation the controller observed, or last generation the controller created?

Reply:
This is the last generation the controller created. Change to lastCreatedGeneration?

Reply:
I think the name is okay, let's just make sure the doc says that it's the last generation created by the controller.